lib/preprocess.js

```diff
@@ -89,15 +92,37 @@ export const preprocess = async ({
   })
 }
 
-export const fetchMeasurementsViaClient = async (web3Storage, cid) => {
-  const res = await web3Storage.get(cid)
-  const files = await res.files()
-  const measurements = JSON.parse(await files[0].text())
-  return measurements
+export const fetchMeasurementsTrustless = async cid => {
+  const res = await fetch(
+    `https://${encodeURIComponent(cid)}.ipfs.w3s.link?format=car`
+  )
+  if (!res.ok) {
+    const msg = `Cannot fetch measurements ${cid}: ${res.status}\n${await res.text()}`
+    throw new Error(msg)
+  }
+  const reader = await CarReader.fromIterable(res.body)
+  const entries = exporter(cid, {
+    async get (cid) {
+      const block = await reader.get(cid)
+      await validateBlock(block)
+      return block.bytes
+    }
+  })
```

Comment:
Awesome that we can stream the content from a gateway and verify it in a streaming way 👏🏻